scripts/fuzz.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -e
+
+function build_fuzz() {
+  PACKAGES=$1
+  
+  for pkg in $PACKAGES; do
+    pushd fuzz/$pkg
+  
+    for file in *.go; do
+      if [[ "$file" == 'fuzz_utils.go' ]]; then
+        continue
+      fi
+  
+      NAME=$(echo $file | sed 's/\.go$//1')
+      echo "Building zip file for $pkg/$NAME"
+      go-fuzz-build -func "Fuzz_$NAME" -o "$pkg-$NAME-fuzz.zip" "github.com/lightningnetwork/lnd/fuzz/$pkg"
+    done
+  
+    popd
+  done
+}
+
+function run_fuzz() {
+  PACKAGES=$1
+  
+  for pkg in $PACKAGES; do
+    pushd fuzz/$pkg
+  
+    for file in *.go; do
+      if [[ "$file" == 'fuzz_utils.go' ]]; then
+        continue
+      fi
+  
+      NAME=$(echo $file | sed 's/\.go$//1')
+      echo "Running fuzzer for $pkg-$NAME-fuzz.zip"
+      go-fuzz -bin="$pkg-$NAME-fuzz.zip" -workdir="$(pwd)" -procs=4
```

Comment:
In the docker build it kills it with the `timeout 10s` function since it's on ubuntu. As is this will run the fuzzer until quit on the first one and not proceed. We want to ideally fuzz for a little bit (some configurable way, cross-compatible amount of time) and then move to the next one. I don't know how to kill something after a couple seconds on my mac (timeout doesnt work on mac)

Reply:
I'm not too aware of cross compatiblity between windows and linux/macOs so my review can really only apply to those.

Reply:
I think it just needs the `-timeout` added (which btw defaults to 10 sec). It could also be an argument for the script such that we can also define it eg. `make fuzz-run timeout=30`

Reply:
`-timeout` is how long one individual mutation that go-fuzz generates (one fuzz test) can run for. So it's restricted in time that way, but the `go-fuzz` main process will still keep mutating indefinitely on the first one.